### PR TITLE
cmd: tweak internal tool lookup to accept more possible locations

### PR DESCRIPTION
**Pull request by bboozzoo**
During development, when one builds the snapd binary and run runs it from some
random location, the internal tool lookup will fall back to using
$(libexecdir)/snapd as root directory for all other binaries.

Attempt to make it easier and try to use the tool from the directory of current
binary, provided the tool exists at that path.

See https://forum.snapcraft.io/t/cant-run-compiled-snapd/10772 for reference.

### CODE REVIEW

File: cmd/cmd_linux.go [hunk 1]
```diff
@@ -144,6 +144,12 @@ func InternalToolPath(tool string) (string, error) {
 			prefix := exe[:idx]
 			return filepath.Join(prefix, "/usr/lib/snapd", tool), nil
 		}
+		// or perhaps some other random location, make sure the tool
```
Comment (pedronis) [inline cmd/cmd_linux.go]: I would be slightly happier if we did this only if idx == -1

File: cmd/cmd_linux.go [hunk 2]
```diff
@@ -144,6 +144,12 @@ func InternalToolPath(tool string) (string, error) {
...
+		// exists there first
+		maybeTool := filepath.Join(filepath.Dir(exe), tool)
+		if osutil.FileExists(maybeTool) {
```
Comment (pedronis) [inline cmd/cmd_linux.go]: we should probably do our own check that maybeTool is actually an executable

Comment (bboozzoo) [reply cmd/cmd_linux.go]: Added a helper in osutil as well.

File: cmd/cmd_linux.go (hunk 1 above)
Comment (bboozzoo) [reply cmd/cmd_linux.go]: Updated.

**Pull request merged**